inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -193,6 +222,56 @@ ConfigType AutoInferencePlugin::GetSupportedConfig(const ConfigType&  config,
     return supportedConfig;
 }
 
+DeviceInformation AutoInferencePlugin::SelectDevice(const std::vector<DeviceInformation>& metaDevices, const std::string& networkPrecision) {
+    if (metaDevices.empty()) {
+        IE_THROW(NotFound) << "No available device to select in AUTO plugin";
+    }
+    if (metaDevices.size() == 1) {
+        return metaDevices.at(0);
+    }
+
+    std::vector<DeviceInformation> CPU;
+    std::vector<DeviceInformation> GPU;
+
+    for (auto& item : metaDevices) {
+        if (item.deviceName.find("CPU") == 0) {
+            CPU.push_back(item);
+            continue;
+        }
+        if (item.deviceName.find("GPU") == 0) {
+            GPU.push_back(item);
+            continue;
+        }
+    }
+
+    if (CPU.empty() && GPU.empty()) {
+        IE_THROW(NotFound) << "No available device found";
+    }
+
+    // Sort GPU by name: GPU.2 > GPU.1 > GPU.0 > GPU, so we always choose the GPU[0] as best device
+    std::sort(GPU.begin(), GPU.end(), [](const DeviceInformation& a, const DeviceInformation& b)->bool{return b.deviceName < a.deviceName;});
```

Comment:
CC @vladimir-paramuzov are you OK with this?

Reply:
so dGPU has a pri over iGPU, but in this ver we do not recognize when there are many dGPUs:
always try the one with highest index first 
If the GPU with highest index fails (e.g. due to no mem) the logic try to load the next in the list (whihc would be next dGPU... etc until we come the iGPU and finally the CPU), etc

This is "greedy" but simple

Reply:
Yes, You're right